### PR TITLE
Add support for Windows

### PR DESCRIPTION
**Pull request by zavitax**
Provide Windows versions of sys/mman.h functions.
Use open() calls suitable for Windows when building on WIN32

### CODE REVIEW

File: c/tree.c [hunk 1]
```diff
@@ -1,16 +1,236 @@
...
+
 #include <fcntl.h>
 #include <errno.h>
 #include <inttypes.h>
 #include <netdb.h>
 #include <stdbool.h>
 #include <stdio.h>
 
+#ifdef WIN32
+	// Source: https://github.com/witwall/mman-win32
```
Comment (oschwald) [inline c/tree.c]: Please split out this code into a separate C file with clear licensing information. It appears that it is under the MIT license.

File: c/tree.c [hunk 2]
```diff
@@ -1,16 +1,236 @@
...
+	int     mlock(const void *addr, size_t len);
+	int     munlock(const void *addr, size_t len);
+
+	#ifndef FILE_MAP_EXECUTE
+	#define FILE_MAP_EXECUTE    0x0020
+	#endif /* FILE_MAP_EXECUTE */
+
+	static int __map_mman_error(const DWORD err, const int deferr)
+	{
+		if (err == 0)
```
Comment (oschwald) [inline c/tree.c]: After splitting out the code, please run `dev-bin/uncrustify-all.sh` on it so that it is formatted similarly to our other code.

Comment (zavitax) [reply c/tree.c]: @oschwald can you provide some guidance for adding another file to the build list?

with regards to dev-bin/uncrustify-all.sh, I'm afraid my environment lacks the tools.

**Comment by oschwald** [comment]
Thanks! This looks great.

**Comment by zavitax** [comment]
@oschwald my pleasure!

**Comment by zavitax** [comment]
@oschwald can you kindly add my name/email to the list of contributors?

**Comment by oschwald** [comment]
The list is generated on release. I just did a release so it should now appear there. Thanks for the reminder!

**Comment by zavitax** [comment]
Thanks! I'm wondering, why didn't it use my full last name?

**Comment by oschwald** [comment]
It gets the names from the Git history. If you want something different to appear, the `.mailmap` file in the distro root can be updated.

**Comment by zavitax** [comment]
I would appreciate if you could update ilya77@gmail.com -> Ilya Melamed

**Comment by oschwald** [comment]
Done in 24887481020aee3d01f3ac1d7902ad403bbe5bfd. Thanks.

**Pull request merged**